tests/migrations/test_migrations_non_atomic/0001_initial.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class AbortMigrationError(Exception):
+    pass
+
+
+def raise_error(apps, schema_editor):
+    # Test atomic operation in non-atomic migration
+    Publisher = apps.get_model('migrations', 'Publisher')
+    Publisher.objects.create(name='RolledBack')
+    raise AbortMigrationError
+
+
+class Migration(migrations.Migration):
+
+    atomic = False
+
+    operations = [
+
+        migrations.CreateModel(
+            "Publisher",
+            [
+                ("id", models.AutoField(primary_key=True)),
+                ("name", models.CharField(max_length=255)),
+            ],
+        ),
+
+        migrations.RunPython(raise_error, atomic=True),
```

Comment:
Related to my comment above: I like the fact that you specify `atomic=True` here, making it explicit that you counter-act the `atomic = False` declared above.